src/HeaderRow.js

```diff
@@ -53,17 +55,23 @@ const HeaderRow = React.createClass({
   },
 
   getHeaderCellType(column) {
+      
     if (column.filterable) {
-      if (this.props.filterable) return HeaderCellType.FILTERABLE;
-    }
-
+         if (this.props.numberfilterable){
+             return HeaderCellType.NUMBERFILTERABLE;
+         }else{
+            if (this.props.filterable) return HeaderCellType.FILTERABLE;
+         }
+    } 
     if (column.sortable) return HeaderCellType.SORTABLE;
-
     return HeaderCellType.NONE;
   },
 
+  getNumberFilterableHeaderCell() {
+      return <NumberFilterableHeaderCell onChange={this.props.onFilterChange} />;
+  },
   getFilterableHeaderCell() {
-    return <FilterableHeaderCell onChange={this.props.onFilterChange} />;
+    return <FilterableHeaderCell onChange={this.props.onFilterChange} />;   
```

Comment:
This is where you should check for the presence of a custom `filterHeaderRenderer`. That way we give the user the flexibility to define a component of their choice instead of giving them a limited choice of predefined renderers.

```
 getFilterableHeaderCell() {
      let FilterHeaderRenderer = this.props.filterHeaderRenderer;
      return FilterHeaderRenderer ? <FilterHeaderRenderer  onChange={this.props.onFilterChange} />: <FilterableHeaderCell onChange={this.props.onFilterChange} />;
}